src/linter.h

```diff
@@ -18,39 +18,77 @@
 #define SRC_LINTER_H_
 
 #include <vector>
+#include <cstdio>
+#include <string>
 
 #include "absl/strings/string_view.h"
 #include "zetasql/base/status.h"
 #include "zetasql/parser/parse_tree_visitor.h"
+#include "zetasql/parser/parse_tree.h"
 #include "zetasql/parser/parser.h"
 #include "zetasql/public/parse_helpers.h"
+#include "zetasql/public/parse_resume_location.h"
 
 namespace zetasql {
 
 namespace linter {
 
-class KeywordExtractor {
+// It gets rule and applies that rule to every ASTnode it visit
+class RuleVisitor : public zetasql::NonRecursiveParseTreeVisitor {
  public:
-    KeywordExtractor(const zetasql::ASTNode *_node,
-                     std::vector<const zetasql::ASTNode *> *_keywords)
-        : node(_node),
-          keywords(_keywords) {}
-    void extract();
+    RuleVisitor(const std::function
+        <absl::Status(const zetasql::ASTNode*, absl::string_view)> &rule,
+                const absl::string_view &sql)
+        : rule_(rule), sql_(sql), result_(absl::OkStatus()) {}
+
+    zetasql_base::StatusOr<VisitResult> defaultVisit(
+        const ASTNode* node) override;
+
+    absl::Status GetResult() { return result_; }
 
  private:
-    const zetasql::ASTNode *node;
-    std::vector<const zetasql::ASTNode *> *keywords;
+    std::function
+        <absl::Status(const zetasql::ASTNode*, absl::string_view)> rule_;
+    absl::string_view sql_;
+    absl::Status result_;
 };
 
-// Checks whether given sql statement is a valid
-// GoogleSql statement
-absl::Status checkStatement(absl::string_view sql);
+//
```

Comment:
missing description?

Reply:
It shouldn't be in there. I am erasing it